query-results/template.yaml

```diff
@@ -350,6 +351,116 @@ Resources:
       QueueName: !Sub ${AWS::StackName}-${Environment}-close-zendesk-ticket-dead-letter-queue
       KmsMasterKeyId: !GetAtt CloseZendeskTicketQueueKmsKey.Arn
 
+  # Query results integration test resources
+  # Writing test data to athena bucket in audit account
+  # Need a separate KMS key here for the integration tests trigger queue because we need to grant access to the audit account
+  # and our main KMS key definitions are in 'core', which is a shared set of definitions across all stacks.
+  WriteTestDataToAthenaBucketQueueKmsKey:
+    Condition: TestEnvironment
+    Type: AWS::KMS::Key
+    Properties:
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: Enable Root access
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action:
+              - kms:*
+            Resource: '*'
+          - !If
+            - LockCrossAccountPermissionsToLambdaRoles
+            # If we are in a non-dev environment so know the lambda role name
+            - Sid: Enable Decrypt access for Audit account write test data to athena bucket lambda
+              Effect: Allow
+              Principal:
+                AWS: '{{resolve:ssm:AuditAccountRootArn}}'
+              Action:
+                - kms:Decrypt
+              Resource: '*'
+              Condition:
+                StringEquals:
+                  'aws:principalArn': !Sub 'arn:aws:iam::{{resolve:ssm:AuditAccountNumber}}:role/txma-ticf-integration-${Environment}-write-test-data-to-athena-bucket-role'
+            # Otherwise allow anything in the dev audit root account
+            - Sid: Enable Decrypt access for Audit dev account
+              Effect: Allow
+              Principal:
+                AWS: '{{resolve:ssm:AuditAccountRootArn}}'
+              Action:
+                - kms:Decrypt
+              Resource: '*'
+          - Sid: Allow AWS Lambda access
+            Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action:
+              - kms:GenerateDataKey
+              - kms:Encrypt
+              - kms:Decrypt
+            Resource: '*'
+
+  WriteTestDataToAthenaBucketQueueKmsKeyAlias:
+    Condition: TestEnvironment
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Sub alias/${AWS::StackName}/${Environment}/write-test-data-to-athena-bucket-queue-kms-key
+      TargetKeyId: !Ref WriteTestDataToAthenaBucketQueueKmsKey
+
+  WriteTestDataToAthenaBucketQueue:
+    Condition: TestEnvironment
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: !Sub ${AWS::StackName}-${Environment}-write-test-data-to-athena-bucket-queue
+      KmsMasterKeyId: !GetAtt WriteTestDataToAthenaBucketQueueKmsKey.Arn
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt WriteTestDataToAthenaBucketDeadLetterQueue.Arn
+        maxReceiveCount: 5
+
+  WriteTestDataToAthenaBucketPolicy:
```

Comment:
This is a bit of a confusing name because it's not a bucket policy, it's a queue policy.

Reply:
Will rename

Reply:
Renamed to WriteTestDataToAthenaBucketQueuePolicy